### PR TITLE
 Fixes #321 - support batch metadata overlay

### LINKED ISSUE #321: Batch metadata overlay: deliverable 1
### Descriptive summary
Existing objects can be edited using a spreadsheet in a batch process form DAMS Manager

### Rationale
We should be able to edit a spreadsheet generated by the batch export tool, then use it to overlay existing object metadata. 

- The spreadsheet should support all columns generated by batch export #301
- Properties and values in the object are replaced with those in the spreadsheet 
- Multiple values in a cell are supported (“ | “ separator)
- Pre-process validation: same as current Excel import
- Reporting similar to current import reporting

[Batch edit specification](https://docs.google.com/document/d/17e921jvvjhFXJA_y1VaDyi9Qmsg_ZNYv4UFSyxACQDM/edit#)

### Related work
#322
#323
#324

### PR DESCRIPTION
**Pull request by lsitu**
Fixes #321

#### Local Checklist
- [x] Tests written and passing locally?
- [x] Code style checked?
- [x] QA-ed locally?
- [x] Rebased with `master` branch?
- [x] Configuration updated (if needed)?
- [ ] Documentation updated (if needed)?

#### What does this PR do?
Support batch metadata overlay with simple objects.

##### Why are we doing this? Any context of related work?
References #321

#### Manual testing steps?
URL: /damsmanager/batchEdit.do

#### Screenshots
Screen Shot 2019-06-25 at 4 33 25 PM

---

#### New ENV variables
References #issuenumber

#### Deployment Instructions

@ucsdlib/developers - please review

### CODE REVIEW

File: WebContent/WEB-INF/jsp/batchEdit.jsp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            $(o).prop('disabled', true);
+        } else {
+            $('input[name="importOption"]').prop('disabled', false);
+            $('input[name="importOption"]').prop('checked', true);
+            $(o).prop('disabled', true);
+        }
+    }
+
+    function reloadPage(){
+        //document.location.href="/damsmanager/batchEdit.do;
```
Comment (mcritchlow) [inline WebContent/WEB-INF/jsp/batchEdit.jsp]: Do we need this commented out still? Or, are you wanting `reloadPage()` to do anything?

File: src/edu/ucsd/library/xdre/tab/ExcelSource.java [hunk 2]
```diff
@@ -660,4 +692,14 @@ private static String normalizeFieldValue(String value, String delimiter)
             value = langElems[0].trim() + delimiter + langElems[1].trim();
         return value;
     }
+
+    /*
+     * Add field dor validation
```
Comment (mcritchlow) [inline src/edu/ucsd/library/xdre/tab/ExcelSource.java]: minor spelling `dor` -> `for`

File: src/edu/ucsd/library/xdre/tab/TabularEditRecord.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    /*
+     * Convert ark to dams4 ARK url
+     * @param ark
+     * @return
+     */
+    private static String toArkUrl(String ark) {
+        if (!(ark.startsWith("http://") || ark.startsWith("https://"))) {
+            String arkUrlBase = Constants.DAMS_ARK_URL_BASE;
+            arkUrlBase += arkUrlBase.endsWith("/") ? "" : "/";
+            return arkUrlBase + "20775/" + ark;
```
Comment (mcritchlow) [inline src/edu/ucsd/library/xdre/tab/TabularEditRecord.java]: In the test, you use `Constants.ARK_ORG`, would that work here as well?

**Comment by mcritchlow** [comment]
@lsitu - it looks like this might need a rebase now. hopefully it's a simple one.

File: WebContent/WEB-INF/jsp/batchEdit.jsp (hunk 1 above)
Comment (lsitu) [reply WebContent/WEB-INF/jsp/batchEdit.jsp]: No. I'll remove it. Thanks.

File: src/edu/ucsd/library/xdre/tab/ExcelSource.java (hunk 2 above)
Comment (lsitu) [reply src/edu/ucsd/library/xdre/tab/ExcelSource.java]: Good catch!

File: src/edu/ucsd/library/xdre/tab/TabularEditRecord.java (hunk 3 above)
Comment (lsitu) [reply src/edu/ucsd/library/xdre/tab/TabularEditRecord.java]: Yes, I think we should use that `Constants.ARK_ORG` instead. Thanks.

**Comment by lsitu** [comment]
@mctchlow I've rebased it with master and cleanup the codes per you review comments (see commit https://github.com/ucsdlib/damsmanager/pull/338/commits/390d9185b0f17b4bc917f70830457da41c888766). Thanks.

**Pull request merged**